packages/nimble-components/src/select/index.ts

```diff
@@ -60,34 +196,203 @@ export class Select extends FoundationSelect implements ErrorPattern {
+        }
+
+        if (e.inputType.includes('deleteContent') || !this.filter.length) {
+            return true;
+        }
+
+        e.stopPropagation();
+        return true;
+    }
+
+    // disabling linting since the override return type should match the parent class return type
+    // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+    public override keydownHandler(e: KeyboardEvent): boolean | void {
+        const key = e.key;
+        if (e.ctrlKey || e.shiftKey) {
+            return true;
+        }
+
+        switch (key) {
+            case keySpace: {
+                if (this.filterMode === FilterMode.none) {
+                    return super.keydownHandler(e);
+                }
+                // when dropdown is open allow user to enter a space for filter text
+                // (calling super method will close dropdown)
+                if (!this.open) {
+                    super.keydownHandler(e);
+                }
+                break;
+            }
+
+            case keyEnter: {
+                this.updateSelectedIndexFromFilteredSet();
+                super.keydownHandler(e);
+                this.focus();
+                break;
+            }
+            case keyEscape: {
+                // clear filteredOptions as call to `super.keydownHandler` will process
+                // "options" and not "_options"
+                this.filteredOptions = [];
+                if (this.committedSelectedOption) {
+                    this.clearSelection();
+                    this.selectedIndex = this._options.indexOf(
+                        this.committedSelectedOption
+                    );
+                }
+                super.keydownHandler(e);
+                // reset 'selected' state after super.keydownHandler is called, otherwise
+                // the selected state doesn't stick.
+                const selectedOption = this._options[this.selectedIndex];
+                if (selectedOption) {
+                    selectedOption.selected = true;
+                }
+                this.focus();
+                break;
+            }
+
+            default: {
+                super.keydownHandler(e);
+            }
+        }
+        return true;
+    }
+
+    // Prevents parent classes from resetting selectedIndex to a positive
+    // value while filtering, which can result in a disabled option being
+    // selected.
+    protected override setSelectedOptions(): void {
+        if (this.open && this.selectedIndex === -1) {
+            return;
+        }
+
+        super.setSelectedOptions();
+    }
+
+    protected override openChanged(
+        prev: boolean | undefined,
+        next: boolean
     ): void {
-        if (this.region && this.control) {
-            this.region.anchorElement = this.control;
+        super.openChanged(prev, next);
+        if (!this.$fastController.isConnected) {
+            return;
+        }
```

Comment:
All the code that ensures the select is in the correct state is skipped when the element is connected. It seems like there is likely a code path involving disconnecting & reconnecting the element and having `open` change while it is disconnected that could result in the element being in an invalid state.

Reply:
I have removed this since I don't believe there are any negative consequences to not focusing the input element on a programmatic setting of open to true.